src/adloader.js

```diff
@@ -18,9 +18,9 @@ const _approvedLoadExternalJSList = [
  * Each unique URL will be loaded at most 1 time.
  * @param {string} url the url to load
  * @param {string} moduleCode bidderCode or module code of the module requesting this resource
- * @param {function} [callback] callback function to be called after the script is loaded.
+ * @param {function} [callback] callback function to be called after the script is loaded
  */
-export function loadExternalScript(url, moduleCode, callback) {
+export function loadExternalScript(url, moduleCode, callback, context) {
```

Comment:
the renderer can load scripts, when the config contains a url property, like;
```js
renderer: {
  url: 'https://cdn.adnxs.com/renderer/video/ANOutstreamVideo.js'
}
```
in which case the script should be loaded in the correct/desired document context.

I'm not sure if calling the callback always would be the best solution, since other modules also make use of this method.
Also this would push the (repeating) logic of caching or not, always to the callback.
However there is 1 advantage to that, since once a script is loaded, it could reference/re initiate the already loaded code in the new context. But this could also be implemented else if desired, like in a pre-render hook. Which would save a url load, if the script needs to be loaded in different document contexts.
Also one could say, you could simply rely on the caching behaviour of the browser.

Reply:
I missed that you are using `context.loadScript` - you are right. I have two suggestions:

- change the name from `context` to something that's clearly a document (like `doc`)
- use `WeakMap` to separate caches by document, e.g. `const caches = new WeakMap(); const _requestCache = caches.get(doc) || caches.set(doc, {}).get(doc)`. It's a smaller change, easier to follow, and marginally more efficient.